pkg/limits/ingest_limits.go

```diff
@@ -385,24 +429,29 @@ func (s *IngestLimits) evictOldStreams(_ context.Context) {
 	for tenant, partitions := range s.metadata {
 		evicted := 0
 		for partitionID, streams := range partitions {
-			for i, stream := range streams {
-				if stream.lastSeenAt < cutoff {
-					// Delete the element without allocating or copying into
-					// a new backing array https://go.dev/wiki/SliceTricks#delete.
-					s.metadata[tenant][partitionID] = append(
-						s.metadata[tenant][partitionID][:i],
-						s.metadata[tenant][partitionID][i+1:]...,
-					)
+			// Create a new slice with only active streams
+			activeStreams := make([]streamMetadata, 0, len(streams))
```

Comment:
Is this better then what we had before? Is there any penalty for re-allocating every time we run an eviction?

Reply:
I find this more readable for functions outside the hot path.